src/EditorFeatures/Core/InlineParameterNameHints/InlineParamNameHintsDataTaggerProvider.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Tagging;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.Tagging;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InlineParameterNameHints;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
+{
+    /// <summary>
+    /// The TaggerProvider that calls upon the service in order to locate the spans and names
+    /// </summary>
+    [Export(typeof(ITaggerProvider))]
+    [ContentType(ContentTypeNames.CSharpContentType)]
+    [TagType(typeof(InlineParamNameHintDataTag))]
+    [Name(nameof(InlineParamNameHintsDataTaggerProvider))]
+    internal class InlineParamNameHintsDataTaggerProvider : AsynchronousTaggerProvider<InlineParamNameHintDataTag>
+    {
+        // [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        [ImportingConstructor]
+        public InlineParamNameHintsDataTaggerProvider(
+            IThreadingContext threadingContext,
+            IAsynchronousOperationListenerProvider listenerProvider,
+            IForegroundNotificationService notificationService)
+            : base(threadingContext, listenerProvider.GetListener(FeatureAttribute.InlineParameterNameHints), notificationService)
+        {
+        }
+
+        protected override ITaggerEventSource CreateEventSource(ITextView textViewOpt, ITextBuffer subjectBuffer)
+        {
+            // TaggerDelay is NearImmediate because we want the renaming and tag creation to be instantaneous
+            return TaggerEventSources.OnTextChanged(subjectBuffer, TaggerDelay.NearImmediate);
```

Comment:
This shouldn't be OnTextChanged, since a change in a different file won't requery it. Should probably be workspace changed or the like.